Log successful AcceptBlockHeader()

Sjors opened the pull request:
Knowing when a header was first seen may help distinguish between a regular reorg and a selfish mining or eclipse attack.

dergoegge commented on src/validation.cpp:
```diff
@@ -3828,6 +3828,7 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
         return state.Invalid(BlockValidationResult::BLOCK_HEADER_LOW_WORK, "too-little-chainwork");
     }
     CBlockIndex* pindex{m_blockman.AddToBlockIndex(block, m_best_header)};
+    LogPrint(BCLog::VALIDATION, "%s: added new block header %s\n", __func__, hash.ToString());
```
```suggestion
    LogPrint(BCLog::VALIDATION, "added new block header %s\n", hash.ToString());
```

We have an option for logging source locations `-logsourcelocations`.

Sjors replied:
Done!

dergoegge reviewed (commented):
Concept ACK

jamesob commented:
Beat me to it! I was just about to open this PR :).

For what it's worth, I think that the right approach here is to make this message `LogPrintf()` and omit it when in initial block download, the rationale being that if something weird happens on the network (like the inspiration for this PR last night), having a lot of data available to investigate would be nice, and most people don't run with verbose logging. Since the additional log volume here is quite minimal (one extra line per block, basically), I don't think there's much cost.

Sjors commented:
I opened #27277 to make `-debug=validation` less noisy, making it a more attractive setting. At least _after_ IBD, since it produces ~4 lines per block: AcceptBlockHeader, NewPoWValidBlock, Pre-allocating and BlockChecked.

fanquake commented:
cc @0xB10C

Sjors committed:
Log successful AcceptBlockHeader()

Knowing when a header was first seen may help distinguish between a regular reorg and a selfish mining or eclipse attack.

Sjors commented on src/validation.cpp:
```diff
@@ -3828,6 +3828,7 @@ bool ChainstateManager::AcceptBlockHeader(const CBlockHeader& block, BlockValida
         return state.Invalid(BlockValidationResult::BLOCK_HEADER_LOW_WORK, "too-little-chainwork");
     }
     CBlockIndex* pindex{m_blockman.AddToBlockIndex(block, m_best_header)};
+    LogPrint(BCLog::VALIDATION, "added new block header %s\n", hash.ToString());
```
Done

Sjors commented:
At this point I prefer #27278, but I'll leave this open in case that PR doesn't make it. This one is (slightly) easier to review.
